src/formatDate.js

```diff
@@ -8,7 +8,40 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateObj = {};
+  const splitDate = date.split(fromFormat[fromFormat.length - 1]);
+
+  const result = [];
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    dateObj[fromFormat[i]] = splitDate[i];
+  }
+
+  if (fromFormat.includes('YY') && toFormat.includes('YYYY')) {
+    if (dateObj.YY) {
+      if (dateObj.YY >= 30) {
+        dateObj.YYYY = '19' + dateObj.YY;
+      } else {
+        dateObj.YYYY = '20' + dateObj.YY;
+      }
+    }
+  }
+
+  if (fromFormat.includes('YYYY') && toFormat.includes('YY')) {
+    if (dateObj.YYYY) {
+      dateObj.YY = dateObj.YYYY.slice(-2);
+    }
+  }
+
+  if (dateObj.hasOwnProperty('YYYY')) {
+    dateObj.YY = dateObj.YYYY.slice(2);
+  }
```

Comment:
This block of code is redundant because the conversion from 'YYYY' to 'YY' is already handled in the previous condition (lines 30-34). You can safely remove this block to avoid unnecessary operations.